ext/storage_sources/gcp_store/test/test_gcp_connection.cpp

```diff
@@ -30,5 +30,338 @@
 #include <catch2/catch.hpp>
 
 #include "gcp_connection.h"
+#include <typeinfo>
 
-TEST_CASE("Testing class gcpConnection", "gcp-connection") {}
+namespace gcs = google::cloud::storage;
+using namespace gcs;
+namespace test_defaults {
+static std::string bucket_name("unit_testing_gcp");
+
+/*
+ * Objects with the prefex pattern "gcptest/*" are deleted after a certain period of time according
+ * to the lifecycle rule on the gcp bucket. Should you wish to make any changes to the prefix
+ * pattern or lifecycle of the object, please speak to the release manager.
+ */
+static std::string obj_prefix("gcptest"); // To be concatenated with a random string.
+} // namespace test_defaults
+
+// Concatenates a random suffix to the prefix being used for the test object keys. Example of
+// generated test prefix: "gcptest/unit/2022-31-01-16-34-10/623843294--".
+static int
+randomize_test_prefix()
```

Comment:
This function is a bit odd.  Would it make more sense to name it `get_test_prefix` and returning the generated string?

Reply:
I agree this function can have a better name, I think it would make more sense to name it `generate_test_prefix`?